### DISCUSSION TITLE
Add DI in Rule

### DISCUSSION THREAD

**Discussion by SenSyIot** [NRules/NRules#278] [Q&A] [answered]
how can  Add DI in  Rules

`public class MyRule : Rule{

private readonly IMyDep _myDep;
public MyRule (IMyDep myDep )
{
       _myDep=myDep;
}
}`

**Comment by snikolayev** [comment, marked as answer]
Sorry for such a late response. Somehow I missed this question. See this wiki page: [Rule activation](https://github.com/NRules/NRules/wiki/Fluent-Rules-Loading#rule-activation). It has a reference implementation for Autofac DI container, but you can implement a rule activator for any container.
